src/Providers/ProviderRegistry.php

```diff
@@ -204,7 +247,20 @@ public function getProviderModel(
 
         // Use static method from ProviderInterface
         /** @var class-string<ProviderInterface> $className */
```

Comment:
This shouldn't be necessary since `resolveProviderClassName` specifies it already.